### PR TITLE
Interrupt test

### PR DESCRIPTION
**Pull request by lorenzoPrimi**
Fixes https://github.com/ooni/probe/issues/1026

## Proposed Changes
- Adding an X button in the running screen
- If the test is interruptable send the signal to the go library
- Empty next tests queue

### CODE REVIEW

File: ooniprobe/Test/Suite/AbstractSuite.h [hunk 1]
```diff
@@ -13,6 +13,7 @@
 @property NSMutableArray *testList;
 @property int measurementIdx;
 @property (nonatomic) UIBackgroundTaskIdentifier backgroundTask;
+@property BOOL interrupted;
```
Comment (bassosimone) [inline ooniprobe/Test/Suite/AbstractSuite.h:16]: I assume the default is `atomic`? Maybe, can you make it explicitly `atomic`?

File: ooniprobe/Test/Test/AbstractTest.h [hunk 2]
```diff
@@ -25,6 +26,7 @@
 @property Settings *settings;
 @property BOOL annotation;
 @property dispatch_queue_t serialQueue;
+@property id<ExperimentTask> task;
```
Comment (bassosimone) [inline ooniprobe/Test/Test/AbstractTest.h:29]: Also this property should be `atomic`

File: ooniprobe/Test/Suite/AbstractSuite.h (hunk 1 above)
Comment (lorenzoPrimi) [reply ooniprobe/Test/Suite/AbstractSuite.h:16]: There is no need to, unless we want `weak`

File: ooniprobe/Test/Test/AbstractTest.h (hunk 2 above)
Comment (lorenzoPrimi) [reply ooniprobe/Test/Test/AbstractTest.h:29]: Same as above

**Pull request merged**